source/rpt/rpt_fem_reconstruction.cc

```diff
@@ -0,0 +1,1080 @@
+    }
+  else if (cost_function_type == Parameters::RPTFEMReconstructionParameters::
+                                   FEMCostFunction::relative)
+    {
+      std::vector<double> denom(detector_size);
+
+      for (unsigned int d = 0; d < experimental_count.size(); ++d)
+        denom[d] = 1 / (experimental_count[d] * experimental_count[d]);
+
+      // Assembling sys_matrix
+      for (unsigned int i = 0; i < dim; ++i)
+        {
+          for (unsigned int j = 0; j < dim; ++j)
+            {
+              for (unsigned int d = 0; d < detector_size; ++d)
+                {
+                  sigma += (-vertex_count[d][0] + vertex_count[d][j + 1]) *
+                           (-vertex_count[d][0] + vertex_count[d][i + 1]) *
+                           denom[d];
+                }
+              sys_matrix.set(i, j, sigma);
+              sigma = 0;
+            }
+        }
+
+      // Assembling sys_rhs
+      for (unsigned int i = 0; i < dim; ++i)
+        {
+          for (unsigned int d = 0; d < detector_size; ++d)
+            {
+              sigma += (vertex_count[d][0] - experimental_count[d]) *
+                       (-vertex_count[d][0] + vertex_count[d][i + 1]) *
+                       denom[d];
+            }
+          sys_rhs[i] = -sigma;
+          sigma      = 0;
+        }
+    }
+
+  // Setup and solve linear system
+  sys_matrix.set_property(LAPACKSupport::general);
+  sys_matrix.compute_lu_factorization();
+  sys_matrix.solve(sys_rhs);
+
+  reference_location = sys_rhs;
+  return reference_location;
+}
+
+
+template <int dim>
+double
+RPTFEMReconstruction<dim>::calculate_reference_location_error(
+  Vector<double> &reference_location,
+  const double &  last_constraint)
+{
+  std::vector<double> err_coordinates(dim + 1, 0);
+  double              norm_error_coordinates = 0;
+
+  // Check if the location is a valid one
+  for (unsigned int i = 0; i < dim; ++i)
+    {
+      if (reference_location[i] > 1)
+        err_coordinates[i] = (reference_location[i] - 1);
+
+      if (reference_location[i] < 0)
+        err_coordinates[i] = (0 - reference_location[i]);
+    }
+
+  // Check the last constraint
+  if (last_constraint < 0)
+    err_coordinates[3] = std::abs(last_constraint);
+  else if (last_constraint > 1)
+    err_coordinates[3] = 1 - last_constraint;
+
+  // Calculate the norm of the error coordinate vector
+  for (const double &error : err_coordinates)
+    {
+      norm_error_coordinates += error * error;
+    }
+
+  return norm_error_coordinates;
+}
+
+
+template <int dim>
+double
+RPTFEMReconstruction<dim>::calculate_cost(
+  const TriaActiveIterator<DoFCellAccessor<dim, dim, false>> &cell,
+  Vector<double> &     reference_location,
+  const double &       last_constraint,
+  std::vector<double> &experimental_count)
+{
+  double cost  = 0;
+  double count = 0;
+
+  if (rpt_parameters.fem_reconstruction_param.fem_cost_function ==
+      Parameters::RPTFEMReconstructionParameters::FEMCostFunction::absolute)
```

Comment:
I agree. Eventually, it would be nice to add these equations to a theory section in the documentation.

Reply:
I've added few equations, but for _sys_matrix_ and _sys_rhs_, since it's made in loops and it's a little packed, Ghazaleh will added them in an appendix of her article and a reference to those equation will be commented.